modules/__tests__/transitionHooks-test.js

```diff
@@ -121,6 +136,35 @@ describe('When a router enters a branch', function () {
     })
   })
 
+  it('calls the route leave hooks when leaving the route', function (done) {
+    const history = createHistory('/news')
+
+    // Stub this function to exercise the code path.
+    history.listenBeforeUnload = () => (() => {})
```

Comment:
would be nice to not have to use implementation specific stubs, and instead transition the history somewhere else after teardown and make sure the added hook isn't called again.